Core/include/Acts/EventData/SpacePointData.ipp

```diff
@@ -0,0 +1,174 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <iostream>
+
+namespace Acts {
+
+inline const float& SpacePointData::x(const std::size_t& idx) const {
```

Comment:
This is nitpicking. Probably having such simple implementation in the .h file would not hurt. And can remove most of boilerplate content here.